build-win32-llvm.txt

```diff
@@ -0,0 +1,20 @@
+[constants]
+winsysroot = '@GLOBAL_SOURCE_ROOT@/winsysroot'
+
+[binaries]
+cpp = ['clang-cl', '/clang:--target=i686-pc-windows-msvc', '/clang:-fuse-ld=lld-link', '/winsysroot', winsysroot]
+strip = 'llvm-strip'
+ar = 'llvm-ar'
+
+[built-in options]
+cpp_args = ['/MT', '-Wno-c++11-narrowing']
```

Comment:
If you're wondering why is this `-Wno-c++11-narrowing` needed, it's to suppress this error:

```
FAILED: src/nvapi64.dll.p/nvapi_drs.cpp.obj 
clang-cl /clang:-fuse-ld=lld-link /winsysroot …/dxvk-nvapi/winsysroot -Isrc/nvapi64.dll.p -Isrc -I../src -I../external/nvapi -I../external/Vulkan-Headers/include -I. /MD /nologo /showIncludes /utf-8 /W3 /EHsc /std:c++20 /permissive- /O2 /Gw -DNOMINMAX -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_WARNINGS /MT /Fdsrc/nvapi64.dll.p/nvapi_drs.cpp.pdb /Fosrc/nvapi64.dll.p/nvapi_drs.cpp.obj /c ../src/nvapi_drs.cpp
In file included from ../src/nvapi_drs.cpp:2:
../external/nvapi/NvApiDriverSettings.c(932,132): error: constant expression evaluates to -1 which cannot be narrowed to type 'NvU32' (aka 'unsigned long') [-Wc++11-narrowing]
  932 |     {SET_VAB_DATA_ID,                      SET_VAB_DATA_STRING,                       6, (NvU32 *)g_valuesSET_VAB_DATA,            SET_VAB_DATA_USE_API_DEFAULTS},
      |                                                                                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
../external/nvapi/NvApiDriverSettings.c(932,132): note: insert an explicit cast to silence this issue
  932 |     {SET_VAB_DATA_ID,                      SET_VAB_DATA_STRING,                       6, (NvU32 *)g_valuesSET_VAB_DATA,            SET_VAB_DATA_USE_API_DEFAULTS},
      |                                                                                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                                                    static_cast<NvU32>(          )
../external/nvapi/NvApiDriverSettings.c(934,132): error: constant expression evaluates to -1769598857 which cannot be narrowed to type 'NvU32' (aka 'unsigned long') [-Wc++11-narrowing]
  934 |     {VSYNCTEARCONTROL_ID,                  VSYNCTEARCONTROL_STRING,                   2, (NvU32 *)g_valuesVSYNCTEARCONTROL,        VSYNCTEARCONTROL_DISABLE},
      |                                                                                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~
../external/nvapi/NvApiDriverSettings.c(934,132): note: insert an explicit cast to silence this issue
  934 |     {VSYNCTEARCONTROL_ID,                  VSYNCTEARCONTROL_STRING,                   2, (NvU32 *)g_valuesVSYNCTEARCONTROL,        VSYNCTEARCONTROL_DISABLE},
      |                                                                                                                                    ^~~~~~~~~~~~~~~~~~~~~~~~
      |                                                                                                                                    static_cast<NvU32>(     )
2 errors generated.
```

But because the error comes from NVAPI headers which we include from a submodule, I thought it would be preferable to suppress this rather than try fixing them.

Reply:
Thanks for the clarification. I think this is fine here. Alternatively, we might wrap https://github.com/jp7677/dxvk-nvapi/blob/master/src/nvapi_drs.cpp#L2 like we do here https://github.com/jp7677/dxvk-nvapi/blob/master/src/nvapi_private.h#L42, but for clang. But I guess that's not worth it, even when moving that include to `nvapi_private.h` to keep the cpp files clean of this stuff (assuming that this doesn't create other issues).